modules/@apostrophecms/modal/ui/apos/components/AposModalShareDraft.vue

```diff
@@ -79,22 +95,95 @@ export default {
         disableHeader: true,
         trapFocus: true
       },
-      disabled: false
+      shareUrl: '',
+      disabled: true
     };
   },
   async mounted() {
     this.modal.active = true;
+    await this.checkUrlprop();
+    await this.getAposShareKey();
   },
   methods: {
-    copy() {
-      console.log('copy');
+    async copy() {
+      await navigator.clipboard.writeText(this.shareUrl);
     },
-    toggle() {
-      console.log('toggle');
+    async toggle() {
       this.disabled = !this.disabled;
+
+      await this.setShareUrl();
+    },
+    async setShareUrl() {
+      try {
+        const { aposShareKey } = await apos.http.post(
+          `${apos.modules[this.doc.type].action}/${this.doc._id}/share`, {
+            busy: true,
+            body: {
+              share: !this.disabled
+            },
+            draft: true
+          }
+        );
+
+        if (this.disabled) {
+          setTimeout(() => {
+            this.shareUrl = '';
+          }, 200);
+          return;
+        }
+
+        if (!aposShareKey) {
+          return this.errorNotif();
+        }
+
+        this.shareUrl = this.generateShareUrl(aposShareKey);
+      } catch {
+        if (this.disabled) {
+          this.shareUrl = '';
+          return;
+        }
+        await this.errorNotif();
+      }
     },
     close() {
       this.modal.showModal = false;
+    },
+    async checkUrlprop() {
+      if (!this.doc._url) {
+        await this.errorNotif();
+      }
+    },
+    async getAposShareKey() {
+      try {
+        const { aposShareKey } = await apos.http.get(
+          `${apos.modules[this.doc.type].action}/${this.doc._id}`, {}
+        );
+
+        if (aposShareKey) {
+          this.disabled = false;
+          this.shareUrl = this.generateShareUrl(aposShareKey);
+        }
+      } catch {
+        await this.errorNotif();
+      }
+    },
+    async errorNotif() {
+      await apos.notify('apostrophe:shareDraftError', {
+        type: 'danger',
+        icon: 'alert-circle-icon',
+        dismiss: true
+      });
+    },
+    generateShareUrl(aposShareKey) {
+      const url = new URL(`${location.origin}${this.doc._url}`);
```

Comment:
should we rely on `getBaseUrl` instead for instances where the backoffice and the website does not use the same domain name?

Reply:
`req.baseUrl` is passed to the frontend, and use instead (in #3798).

But I'm not sure that `baseUrl` will take the front-office URL.
Beside, I think that we should use `baseUrlWithPrefix`, but it's undefined.

I'll dig more on that, but first let's ask @boutell if there's a "right" way to get the front-office domain/base url!